library/ssl_tls.c

```diff
@@ -6984,10 +7117,17 @@ static int ssl_tls12_populate_transform( mbedtls_ssl_transform *transform,
         transform->maclen = mac_key_len;
 
         /* IV length */
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        if( ssl_mode == MBEDTLS_SSL_MODE_STREAM )
+            transform->ivlen = 0;
+        else
+            transform->ivlen = PSA_CIPHER_IV_LENGTH( key_type, alg );
```

Comment:
Isn't this just
```suggestion
        transform->ivlen = PSA_CIPHER_IV_LENGTH( key_type, alg );
```
? When `ssl_mode == MBEDTLS_SSL_MODE_STREAM` (so null cipher — and this would be true with RC4 as well), `PSA_CIPHER_IV_LENGTH( key_type, alg )` is 0.

Reply:
I was not sure about this because alg would be set to `MBEDTLS_SSL_NULL_CIPHER` which is an MbedTLSism and I wasn't sure if was safe to use this value against the PSA API.
But it seems it's defined as the same as `PSA_ALG_CATEGORY_CIPHER` and `PSA_CIPHER_IV_LENGTH` will return 0 since it matches no ALGs.

Reply:
Is it guaranteed behaviour that `PSA_CIPHER_IV_LENGTH()` returns 0 when `alg` is not a valid cipher (or at least when it's `PSA_ALG_CATEGORY_CIPHER`)?

Reply:
In the PSA spec, it's at least guaranteed to return 0 for 0 and for a cipher that doesn't use an IV. Our implementation also guarantees “If the key type or cipher algorithm is not recognized, or the parameters are incompatible, return 0”.

(The PSA spec allows other values in the not-recognized case — for example an implementation that only supports ECB and no other non-authenticated mode could always return 0, and an implementation that only supports CBC could always return the block size. But our TLS code doesn't need to accommodate other ciphers. And anyway the only potential non-valid input is 0 for a null cipher, and that's guaranteed to yield 0 since it isn't a cipher algorithm.)

Reply:
Ok, thanks!